jscomp/syntax/src/res_core.ml

```diff
@@ -3923,6 +3941,30 @@ and parseListExpr ~startPos p =
             loc))
       [(Asttypes.Nolabel, Ast_helper.Exp.array ~loc listExprs)]
 
+and parseDictExpr ~startPos p =
+  let exprs =
+    parseCommaDelimitedRegion ~grammar:Grammar.DictRows ~closing:Rbrace
+      ~f:parseDictExprRow p
+  in
+  let loc = mkLoc startPos p.prevEndPos in
+  let toKeyValuePair recordItem =
+    match recordItem with
+    | {Location.txt = Longident.Lident key}, valueExpr ->
+      Some
+        (Ast_helper.Exp.tuple ~loc
+           [Ast_helper.Exp.constant ~loc (Pconst_string (key, None)); valueExpr])
```

Comment:
Same question here. I think we want this location to only cover the `"key"` part.